### PR TITLE
Components: Add Portal

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Adds a Portal component that wraps reakit's Portal with the ThemeProvider so that themes are correctly applied to portalled elements.

This PR also updates G2 to get fixes to the ThemeProvider props as well as the contextConnect function:

https://github.com/ItsJonQ/g2/pull/263
https://github.com/ItsJonQ/g2/pull/264

## How has this been tested?
Unit tests pass

## Types of changes
New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by ItsJonQ** [comment]
@sarayourfriend Hmm! I'm not sure why the `ControlGroup` changes are here 🤔 . I don't think we need them for `Portal`.
If they are, I'm guessing the ControlGroup update (https://github.com/WordPress/gutenberg/pull/28982) should resolve things :)

**Comment by sarayourfriend** [comment]
@ItsJonQ I have... no idea how that happened... I'll fix it right away, sorry about that.

**Comment by ItsJonQ** [comment]
@sarayourfriend No worries :D! I also updated the Portal test

**Comment by ItsJonQ** [comment]
Will add a quick README now

File: packages/components/src/ui/portal/test/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+			</>
+		);
+
+		// This is because the `div.element` is portal'ed outside of the wrapping `.container`
+		expect(
+			document.querySelector( 'body .container .element' )
+		).toBeFalsy();
+
+		// Instead, it is rendered at the `body` level.
+		expect( document.querySelector( 'body .element' ) ).toBeTruthy();
```
Comment (sarayourfriend) [inline packages/components/src/ui/portal/test/index.js:31]: 🎉

**Review by ItsJonQ — APPROVED** [review]
🚀 from me!!

**Comment by alexstine** [comment]
@sarayourfriend Could this PR be related to the error I reported here?

https://wordpress.slack.com/archives/C02QB2JS7/p1613690151252200

Builds are failing on latest master for me.

**Comment by alexstine** [comment]
Sorry. Never mind. Just required a fresh npm install.

**Pull request merged**